tests/test_dynamo/test_user.py

```diff
@@ -1,73 +1,253 @@
+            unittest.mock.patch('dynamo.user._create_user') as mock_create_user:
+        mock_get_current_month.return_value = '2024-02'
+        mock_create_user.return_value = 'create_user_return_value'
+
+        user = dynamo.user.get_or_create_user('foo')
+
+        mock_get_current_month.assert_called_once_with()
+        mock_create_user.assert_called_once_with(
+            user_id='foo',
+            default_credits=Decimal(25),
+            current_month='2024-02',
+            users_table=tables.users_table,
+        )
+
+    assert user == 'create_user_return_value'
+
+
+def test_create_user(tables):
+    user = dynamo.user._create_user(
+        user_id='foo',
+        default_credits=Decimal(25),
+        current_month='2024-02',
+        users_table=tables.users_table
+    )
+
+    assert user == {'user_id': 'foo', 'remaining_credits': Decimal(25), 'month_of_last_credits_reset': '2024-02'}
+    assert tables.users_table.scan()['Items'] == [user]
+
+
+def test_create_user_already_exists(tables):
+    tables.users_table.put_item(Item={'user_id': 'foo'})
+
+    with pytest.raises(dynamo.user.DatabaseConditionException):
+        dynamo.user._create_user(
+            user_id='foo',
+            default_credits=Decimal(25),
+            current_month='2024-02',
+            users_table=tables.users_table
+        )
+
+    assert tables.users_table.scan()['Items'] == [{'user_id': 'foo'}]
+
+
+def test_reset_credits(tables, monkeypatch):
+    monkeypatch.setenv('RESET_CREDITS_MONTHLY', 'yes')
+
+    original_user_record = {
+        'user_id': 'foo', 'remaining_credits': Decimal(10), 'month_of_last_credits_reset': '2024-01'
+    }
+    tables.users_table.put_item(Item=original_user_record)
+
+    user = dynamo.user._reset_credits_if_needed(
+        user=original_user_record,
+        default_credits=Decimal(25),
+        current_month='2024-02',
+        users_table=tables.users_table,
+    )
+
+    assert user == {'user_id': 'foo', 'remaining_credits': Decimal(25), 'month_of_last_credits_reset': '2024-02'}
+    assert tables.users_table.scan()['Items'] == [user]
+
+
+def test_reset_credits_no_reset(tables, monkeypatch):
+    monkeypatch.setenv('RESET_CREDITS_MONTHLY', 'no')
+
+    original_user_record = {
+        'user_id': 'foo', 'remaining_credits': Decimal(10), 'month_of_last_credits_reset': '2024-01'
     }
+    tables.users_table.put_item(Item=original_user_record)
 
-    user2 = dynamo.user.create_user('user2')
-    assert user2 == {
-        'user_id': 'user2',
-        'remaining_credits': 25,
+    user = dynamo.user._reset_credits_if_needed(
+        user=original_user_record,
+        default_credits=Decimal(25),
+        current_month='2024-02',
+        users_table=tables.users_table,
+    )
+
+    assert user == {'user_id': 'foo', 'remaining_credits': Decimal(10), 'month_of_last_credits_reset': '2024-01'}
+    assert tables.users_table.scan()['Items'] == [user]
```

Comment:
I tend to prefer using literal values as much as possible in tests, but I think re-using `original_user_record ` makes the asserts more expressive
```suggestion
    assert user == original_user_record
    assert tables.users_table.scan()['Items'] == [original_user_record]
```